gwpy/segments/flag.py

```diff
@@ -1569,6 +1581,25 @@ def populate(self, source=DEFAULT_SEGMENT_SERVER,
                     self[key].active &= segments
         return self
 
+    def copy(self, deep=False):
+        """Build a copy of this dictionary.
+
+        Parameters
+        ----------
+        deep : `bool`, optional, default: `False`
+            perform a deep copy of the original dictionary with a fresh 
+            memory address
+
+        Returns
+        -------
+        flag2 : `DataQualityFlag`
+            a copy of the original dictionary
+        """
+        if deep:
+            return deepcopy(self)
+        else:
+            return shallowcopy(self)
```

Comment:
@tjma12, I would prefer to keep the default open using `super`, just in case the `OrderedDict` does something more than just `shallowcopy`; also, you don't need the `else` here:

```python
        if deep:
            return deepcopy(self)
        return super(DataQualityDict, self).copy()
```